src/backend/columnar/columnar_tableam.c

```diff
@@ -1863,9 +1863,7 @@ ColumnarProcessUtility(PlannedStmt *pstmt,
 									   GetCreateIndexRelationLockMode(indexStmt));
 		if (rel->rd_tableam == GetColumnarTableAmRoutine())
 		{
-			/* for now, we don't support index access methods other than btree & hash */
-			if (strncmp(indexStmt->accessMethod, "btree", NAMEDATALEN) != 0 &&
-				strncmp(indexStmt->accessMethod, "hash", NAMEDATALEN) != 0)
+			if (!ColumnarSupportsIndexAM(indexStmt->accessMethod))
 			{
 				ereport(ERROR, (errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
 								errmsg("only btree and hash indexes are supported on "
```

Comment:
> A hint could contain the list of supported access methods. The biggest problem I have with listing the supported methods is that these change over time which cause many tests to change their output once we start supporting them.

Agree, so I think we should not list supported / unsupported indexAM's anyway.

> The detail could contain the access method found that is unsupported and why - eg. gist is unsupported as it relies on a bitmap index scan.

I'm more inclined to not give such details for simplicity, what do you think @thanodnl ?

Reply:
Yeah, makes sense. What indexes are and are not supported can be touched upon in docs.
Anyway, now with the new guidance from product we will push for all index support anyway, so this might actually be removed anyway.